core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -779,6 +779,22 @@ public void testAddRequiredColumnCaseInsensitive() {
         .hasMessage("Cannot add column, name already exists: ID");
   }
 
+  @Test
```

Comment:
Could we also add an explicit test where the initial schema has (id int, data string) and then we go and try and add `DATA` with case insensitivity? I know the logic will also catch this case but I think it's good to have that explicit test since it verifies that our update logic is holistically evaluating the schema instead of just in incremental fashion.

Reply:
I see, we already had an additional incremental validation in `internalAddColumn` which was catching the case where a duplicate field is added to a schema already containing that field in a case-insensitive way. Cool, no need to duplicate the same test then.